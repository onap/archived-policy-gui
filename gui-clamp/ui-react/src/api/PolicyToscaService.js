/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019, 2021 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class PolicyToscaService {
  static getToscaPolicyModels() {
    return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels', { method: 'GET', credentials: 'same-origin' })
      .then(function (response) {
        console.debug("getToscaPolicyModels response received: ", response.status);
        if (response.ok) {
          return response.json();
        } else {
          console.error("getToscaPolicyModels query failed");
          return {};
        }
      })
      .catch(function (error) {
        console.error("getToscaPolicyModels error received", error);
        return {};
      });
  }

  static getToscaPolicyModelYaml(policyModelType, policyModelVersion) {
        return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/yaml/' + policyModelType + "/" + policyModelVersion, {
            method: 'GET',
            credentials: 'same-origin'
        })
            .then(function (response) {
                console.debug("getToscaPolicyModelYaml response received: ", response.status);
                if (response.ok) {
                    return response.json();
                } else {
                    console.error("getToscaPolicyModelYaml query failed");
                    return "";
                }
            })
            .catch(function (error) {
                console.error("getToscaPolicyModelYaml error received", error);
                return "";
            });
 }

 static getToscaPolicyModel(policyModelType, policyModelVersion) {
         return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/' + policyModelType + "/" + policyModelVersion, {
             method: 'GET',
             credentials: 'same-origin'
         })
             .then(function (response) {
                 console.debug("getToscaPolicyModel response received: ", response.status);
                 if (response.ok) {
                     return response.json();
                 } else {
                     console.error("getToscaPolicyModel query failed");
                     return {};
                 }
             })
             .catch(function (error) {
                 console.error("getToscaPolicyModel error received", error);
                 return {};
             });
  }

  static createPolicyModelFromToscaModel(jsonData) {
       return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels', {
           method: 'POST',
           credentials: 'same-origin',
           headers: {
             "Content-Type": "a",
           },
           body: JSON.stringify(jsonData)
         })
         .then(function(response) {
           console.debug("createPolicyModelFromToscaModel response received: ", response.status);
           if (response.ok) {
             var message = {
               status: response.status,
               message: 'Tosca Policy Model successfully uploaded'
             };
             return message;
           } else {
             console.error("createPolicyModelFromToscaModel failed");
             return response.text();
           }
         })
         .catch(function(error) {
           console.error("createPolicyModelFromToscaModel error received", error);
           return "";
         });
     }

     static updatePolicyModelTosca(policyModelType, policyModelVersion, jsonData) {
         return fetch(window.location.pathname + 'restservices/clds/v2/policyToscaModels/' + policyModelType + '/' + policyModelVersion, {
             method: 'PUT',
             credentials: 'same-origin',
             headers: {
               "Content-Type": "a",
             },
             body: JSON.stringify(jsonData)
           })
           .then(function(response) {
             console.debug("updatePolicyModelTosca response received: ", response.status);
             if (response.ok) {
               var message = {
                 status: response.status,
                 message: 'Tosca Policy Model successfully uploaded'
               };
               return message;
             } else {
               console.error("updatePolicyModelTosca failed");
               return response.text();
             }
           })
           .catch(function(error) {
             console.error("updatePolicyModelTosca error received", error);
             return "";
           });
       }
}
