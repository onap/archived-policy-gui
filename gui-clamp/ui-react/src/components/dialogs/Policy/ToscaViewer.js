/*-
 * ============LICENSE_START=======================================================
 * ONAP POLICY-CLAMP
 * ================================================================================
 * Copyright (C) 2021 AT&T Intellectual Property. All rights
 *                             reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

import React from 'react'
import PolicyToscaService from '../../../api/PolicyToscaService';
import styled from 'styled-components';
import Button from 'react-bootstrap/Button';

const JsonEditorDiv = styled.div`
    margin-top: 20px;
    background-color: ${props => props.theme.toscaTextareaBackgroundColor};
    text-align: justify;
    font-size: ${props => props.theme.toscaTextareaFontSize};
    width: 100%;
    height: 30%;
`

export default class ToscaViewer extends React.Component {

   state = {
        toscaData: this.props.toscaData,
        yamlPolicyTosca: this.getToscaModelYamlFor(this.props.toscaData),
   }

   constructor(props, context) {
        super(props, context);
        this.getToscaModelYamlFor = this.getToscaModelYamlFor.bind(this);
   }

   getToscaModelYamlFor(toscaData) {
        PolicyToscaService.getToscaPolicyModelYaml(toscaData["policyModelType"], toscaData["version"]).then(respYamlPolicyTosca => {
            this.setState({
                            yamlPolicyTosca: respYamlPolicyTosca,
                        })
        });
   }

   render() {
       return (
                    <JsonEditorDiv>
                        <pre>{this.state.yamlPolicyTosca}</pre>
                        <Button variant="secondary" title="Create a new policy version from the defined parameters"
                            onClick={this.handleCreateNewVersion}>Create New Version</Button>
                    </JsonEditorDiv>
       );
   }
}