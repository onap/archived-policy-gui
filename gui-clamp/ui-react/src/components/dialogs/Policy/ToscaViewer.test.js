/*-
 * ============LICENSE_START=======================================================
 * ONAP POLICY-CLAMP
 * ================================================================================
 * Copyright (C) 2021 AT&T Intellectual Property. All rights
 *                             reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */
import React from 'react';
import ToscaViewer from './ToscaViewer';
import { shallow, mount } from 'enzyme';
import PolicyToscaService from '../../../api/PolicyToscaService';

describe('Verify ToscaViewer', () => {
  const fs = require('fs');

  let toscaYaml = fs.readFileSync('src/components/dialogs/Policy/toscaData.test.yaml', { encoding: 'utf8', flag: 'r' })

  const toscaData = {
    "policyModelType": "onap.policies.controlloop.Guard",
    "version": "1.0.0",
    "policyAcronym": "Guard",
    "createdDate": "2021-04-09T02:29:31.407356Z",
    "updatedDate": "2021-04-09T02:29:31.407356Z",
    "updatedBy": "Not found",
    "createdBy": "Not found",
    "tableData": {
      "id": 0
    }
  };

  it('Test the render method', async () => {
    PolicyToscaService.getToscaPolicyModelYaml = jest.fn().mockImplementation(() => {
      return Promise.resolve(toscaYaml);
    });
    const component = shallow(<ToscaViewer toscaData={ toscaData }/>);
    await PolicyToscaService.getToscaPolicyModelYaml();
    expect(component).toMatchSnapshot();
  });
});
