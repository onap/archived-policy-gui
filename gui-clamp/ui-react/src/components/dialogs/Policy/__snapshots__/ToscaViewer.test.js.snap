// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify ToscaViewer Test the render method 1`] = `
<styled.div>
  <pre>
    tosca_definitions_version: tosca_simple_yaml_1_1_0
policy_types:
  onap.policies.controlloop.Guard:
    properties: {
    }
    name: onap.policies.controlloop.Guard
    version: 1.0.0
    derived_from: tosca.policies.Root
    metadata: {
    }
    description: Guard Policies for Control Loop Operational Policies
name: ToscaServiceTemplateSimple
version: 1.0.0

  </pre>
  <Button
    active={false}
    disabled={false}
    title="Create a new policy version from the defined parameters"
    variant="secondary"
  >
    Create New Version
  </Button>
</styled.div>
`;
